### DISCUSSION TITLE
Klassendiagramme

### DISCUSSION THREAD

**Discussion by milack68** [Webmasters-Europe/codemeetup-forum#24] [Show and tell] [answered]
Hier ein Vorschlag zur Umsetzung der User-, Role- und Permission Migrationen:
Allenfalls könnte man das User-Model noch teilen in ein User und ein Profile-Model.

![UserRolePermissionKlassendiagramm](https://user-images.githubusercontent.com/45495538/107233955-6f619d80-6a23-11eb-924b-bac1b99e0452.png)

**Comment by Cosnavel** [comment, marked as answer]
Würde ggf. das Package von Spatie verwenden. Was sagst du dazu?

[https://github.com/spatie/laravel-permission](https://github.com/spatie/laravel-permission)

**Reply by milack68** [reply]
Schaut super aus, dieses Package. Es gibt da auch eine gute Dokumentation: https://spatie.be/docs/laravel-permission/v4/introduction

**Comment by milack68** [comment]
Hier eine kleine Anregung für die heutige Session (noch ohne die Relationships):
Idee: es gibt ein oder mehrere Foren, jedes davon hat Themen, dieses wiederum hat Post und diese haben PostReplies.

![ForumKlassendiagramm](https://user-images.githubusercontent.com/45495538/107845616-07250a00-6ddd-11eb-964f-2dc457dfe56f.png)

**Comment by milack68** [comment]
@Cosnavel Wie könnten wir hier die Umsetzung der Models, Migrations und Controllers sinnvoll in einzelnen Issues formulieren? Bräuchte hier einen ersten Tipp...

**Reply by Cosnavel** [reply]
Würde inkrementell vorgehen. Zuerst ein Post. Dann ein Thema, welches einem Post zugewiesen werden kann. Denke mit dem Thema ist quasi eine Kategorie gemeint. Ein Kommentar oder Antwortfunktion kannst auch gut als einzelnen Feature Request behandeln. 
Was ist mit Forum gemeint? Verstehe den Sinn der Tabelle gerade noch nicht. Kannst du mich da aufklären?

**Reply by milack68** [reply]
Also mit Forum habe ich folgende Situation gemeint. Nehmen wir unsere "Lieblingsfirma" Apple her.
Forum wären zB Mac oder IPhone oder IPad
Thema wären dann Mac Mini, Macbook Air, IPhone 12
Post wäre dann der jeweilige Beitrag oder die Frage (zB Wie mache ich einen Hardreset ?, Welche "must-have" Apps brauche ich?)
Und PostReplys wären dann die Antworten auf die Posts

Ist das zu komplex oder kompliziert?

**Comment by milack68** [comment]
Wir sollten darüber diskutieren wie unser Forum aufgebaut ist, damit wir dann beginnen können die Hauptfunktionen umsetzen können.
Ein Vorschlag findet sich in den vorigen Posts.

Alternativ könnte man zB auch das Model Forum weglassen und gleich mit Themen beginnen. Die Themen können aber auch Label oder Tags sein. Bitte um eure Meinungen dazu.
